apidump/summary.go

```diff
@@ -0,0 +1,177 @@
+}
+
+// Dumps packet counters for packets captured and sent to the Akita backend.
+// If the debug flag is set, also prints packets taht were captured but not
+// sent to the backend.
+func (s *Summary) PrintPacketCounts() {
+	if len(s.NegationFilters) == 0 {
+		DumpPacketCounters(printer.Stderr.Infof, s.Interfaces, s.FilterSummary, nil, true)
+	} else {
+		DumpPacketCounters(printer.Stderr.Infof, s.Interfaces, s.FilterSummary, s.NegationSummary, true)
+	}
+
+	if viper.GetBool("debug") {
+		if s.NumUserFilters > 0 {
+			printer.Stderr.Debugf("+++ Counts before allow and exclude filters and sampling +++\n")
+			DumpPacketCounters(printer.Stderr.Debugf, s.Interfaces, s.PrefilterSummary, nil, false)
+		}
+	}
+}
+
+// Prints warnings based on packet capture behavior, such as not capturing
+// any packets, capturing packets but failing to parse them, etc.
+func (s *Summary) PrintWarnings() {
+	// Report on recoverable error counts during trace
+	if pcap.CountNilAssemblerContext > 0 || pcap.CountNilAssemblerContextAfterParse > 0 || pcap.CountBadAssemblerContextType > 0 {
+		printer.Stderr.Infof("Detected packet assembly context problems during capture: %v empty, %v bad type, %v empty after parse",
+			pcap.CountNilAssemblerContext,
+			pcap.CountBadAssemblerContextType,
+			pcap.CountNilAssemblerContextAfterParse)
+		printer.Stderr.Infof("These errors may cause some packets to be missing from the trace.")
+	}
+
+	// Check summary to see if the trace will have anything in it.
+	totalCount := s.FilterSummary.Total()
+	if totalCount.HTTPRequests == 0 && totalCount.HTTPResponses == 0 {
+		// TODO: recognize TLS handshakes and count them separately!
+		if totalCount.TCPPackets == 0 {
+			if s.CapturingNegation && s.NegationSummary.Total().TCPPackets == 0 {
+				printer.Stderr.Infof("%s\n", printer.Color.Yellow("Did not capture any TCP packets during the trace."))
+				printer.Stderr.Infof("%s\n", printer.Color.Yellow("This may mean the traffic is on a different interface, or that"))
+				printer.Stderr.Infof("%s\n", printer.Color.Yellow("there is a problem sending traffic to the API."))
+			} else {
+				printer.Stderr.Infof("%s\n", printer.Color.Yellow("Did not capture any TCP packets matching the filter."))
+				printer.Stderr.Infof("%s\n", printer.Color.Yellow("This may mean your filter is incorrect, such as the wrong TCP port."))
+			}
+		} else if totalCount.Unparsed > 0 {
+			printer.Stderr.Infof("Captured %d TCP packets total; %d unparsed TCP segments.\n",
+				totalCount.TCPPackets, totalCount.Unparsed)
+			printer.Stderr.Infof("%s\n", printer.Color.Yellow("This may mean you are trying to capture HTTPS traffic."))
+			printer.Stderr.Infof("See https://docs.akita.software/docs/proxy-for-encrypted-traffic\n")
+			printer.Stderr.Infof("for instructions on using a proxy, or generate a HAR file with\n")
+			printer.Stderr.Infof("your browser as described in\n")
+			printer.Stderr.Infof("https://docs.akita.software/docs/collect-client-side-traffic-2\n")
```

Comment:
Is this outdated?

Reply:
This is still valid, although the docs could use some updated screenshots.